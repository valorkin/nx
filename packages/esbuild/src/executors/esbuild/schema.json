{
  "title": "esbuild (experimental)",
  "description": "Bundle a package for different platforms. Note: declaration (*.d.ts) file are not currently generated.",
  "cli": "nx",
  "type": "object",
  "properties": {
    "main": {
      "type": "string",
      "description": "The path to the entry file, relative to project.",
      "alias": "entryFile",
      "x-completion-type": "file",
      "x-completion-glob": "**/*@(.js|.ts)"
    },
    "outputPath": {
      "type": "string",
      "description": "The output path of the generated files.",
      "x-completion-type": "directory"
    },
    "outputFileName": {
      "type": "string",
      "description": "Name of the main output file. Defaults same basename as 'main' file."
    },
    "tsConfig": {
      "type": "string",
      "description": "The path to tsconfig file.",
      "x-completion-type": "file",
      "x-completion-glob": "tsconfig.*.json"
    },
    "project": {
      "type": "string",
      "description": "The path to package.json file."
    },
    "additionalEntryPoints": {
      "type": "array",
      "description": "List of additional entry points.",
      "items": { "type": "string" },
      "default": []
    },
    "format": {
      "type": "array",
      "description": "List of module formats to output. Defaults to matching format from tsconfig (e.g. CJS for CommonJS, and ESM otherwise).",
      "alias": "f",
      "items": { "type": "string", "enum": ["esm", "cjs"] }
    },
    "watch": {
      "type": "boolean",
      "description": "Enable re-building when files change.",
      "default": false
    },
    "assets": {
      "type": "array",
      "description": "List of static assets.",
      "default": [],
      "items": {
        "$ref": "#/definitions/assetPattern"
      }
    },
    "deleteOutputPath": {
      "type": "boolean",
      "description": "Remove previous output before build.",
      "alias": "clean",
      "default": true
    },
    "external": {
      "type": "array",
      "description": "Mark one or more module as external. Can use * wildcards, such as '*.png'.",
      "items": {
        "type": "string"
      }
    },
    "format": {
      "type": "array",
      "description": "Set the output format(s).",
      "alias": "f",
      "items": {
        "type": "string",
        "enum": ["esm", "cjs"]
      },
      "default": ["esm"]
    },
    "outputHashing": {
      "type": "string",
      "description": "Define the output filename cache-busting hashing mode.",
      "default": "none",
      "enum": ["none", "all"]
    },
    "metafile": {
      "type": "boolean",
      "description": "Generate a meta.json file in the output folder that includes metadata about the build. This file can be analyzed by other tools.",
      "default": false
    },
    "sourcemap": {
      "type": "boolean",
      "alias": "sourceMap",
      "description": "Generate sourcemap.",
      "default": false
    },
    "minify": {
      "type": "boolean",
      "description": "Minifies outputs.",
      "default": false
    },
    "platform": {
      "type": "string",
      "description": "Platform target for outputs.",
      "enum": ["browser", "node", "neutral"],
      "default": "node"
    },
    "target": {
      "type": "string",
      "description": "The environment target for outputs.",
      "default": "esnext"
    },
    "skipTypeCheck": {
      "type": "boolean",
      "description": "Skip type-checking via TypeScript. Skipping type-checking speeds up the build but type errors are not caught.",
      "default": false
    },
    "updateBuildableProjectDepsInPackageJson": {
      "type": "boolean",
      "description": "Update buildable project dependencies in `package.json`.",
      "default": true
    },
    "buildableProjectDepsInPackageJsonType": {
      "type": "string",
      "description": "When `updateBuildableProjectDepsInPackageJson` is `true`, this adds dependencies to either `peerDependencies` or `dependencies`.",
      "enum": ["dependencies", "peerDependencies"],
      "default": "peerDependencies"
    },
    "esbuildOptions": {
      "type": "object",
      "description": "Additional options to pass to esbuild. See https://esbuild.github.io/api/.",
      "additionalProperties": true
    }
  },
  "required": ["tsConfig", "main", "outputPath"],
  "definitions": {
    "assetPattern": {
      "oneOf": [
        {
          "type": "object",
          "properties": {
            "glob": {
              "type": "string",
              "description": "The pattern to match."
            },
            "input": {
              "type": "string",
              "description": "The input directory path in which to apply `glob`. Defaults to the project root."
            },
            "output": {
              "type": "string",
              "description": "Relative path within the output folder."
            },
            "ignore": {
              "description": "An array of globs to ignore.",
              "type": "array",
              "items": {
                "type": "string"
              }
            }
          },
          "additionalProperties": false,
          "required": ["glob", "input", "output"]
        },
        {
          "type": "string"
        }
      ]
    }
  },
  "additionalProperties": true,
  "examplesFile": "../../../docs/esbuild-examples.md"
}
