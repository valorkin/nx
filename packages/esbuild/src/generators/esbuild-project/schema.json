{
  "$schema": "http://json-schema.org/schema",
  "$id": "NxEsBuildProject",
  "cli": "nx",
  "title": "Add esbuild configuration to a project",
  "description": "Add esbuild configuration to a project.",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-dropdown": "project",
      "x-prompt": "What is the name of the project to set up a esbuild for?"
    },
    "main": {
      "type": "string",
      "description": "Path relative to the workspace root for the main entry file. Defaults to `<project-root>/src/main.ts` or `<project-root>src/index.ts`, whichever is found.",
      "alias": "entryFile"
    },
    "tsConfig": {
      "type": "string",
      "description": "Path relative to the workspace root for the tsconfig file to build with. Defaults to `<project-root>/tsconfig.app.json` or `<project-root>/tsconfig.lib.json`, whichever is found."
    },
    "skipFormat": {
      "description": "Skip formatting files.",
      "type": "boolean",
      "default": false
    },
    "skipPackageJson": {
      "type": "boolean",
      "default": false,
      "description": "Do not add dependencies to `package.json`."
    },
    "skipValidation": {
      "type": "boolean",
      "default": false,
      "description": "Do not perform any validation on existing project."
    },
    "importPath": {
      "type": "string",
      "description": "The library name used to import it, like `@myorg/my-awesome-lib`."
    },
    "platform": {
      "type": "string",
      "description": "Platform target for outputs.",
      "enum": ["browser", "node", "neutral"],
      "default": "node"
    }
  },
  "required": [],
  "examplesFile": "../../../docs/esbuild-project-examples.md"
}
