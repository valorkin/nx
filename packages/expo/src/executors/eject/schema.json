{
  "cli": "nx",
  "$id": "NxExpoEject",
  "$schema": "http://json-schema.org/schema",
  "title": "Expo Eject",
  "description": "Create native iOS and Android project files",
  "type": "object",
  "properties": {
    "install": {
      "type": "boolean",
      "description": "Install CocoaPods.",
      "default": true
    },
    "platform": {
      "type": "string",
      "description": "Platforms to sync",
      "default": "all",
      "examples": ["ios", "android", "all"]
    }
  }
}
