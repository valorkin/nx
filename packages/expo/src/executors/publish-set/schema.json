{
  "cli": "nx",
  "$id": "NxExpoPublishSet",
  "$schema": "http://json-schema.org/schema",
  "title": "Set Publish Channel for Expo",
  "description": "Specify the channel to serve a published release",
  "type": "object",
  "properties": {
    "releaseChannel": {
      "type": "string",
      "description": "The release channel to publish to."
    },
    "publishId": {
      "type": "string",
      "description": "The id of the published release to serve from the channel."
    }
  },
  "required": ["releaseChannel", "publishId"]
}
