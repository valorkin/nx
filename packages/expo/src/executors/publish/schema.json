{
  "cli": "nx",
  "$id": "NxExpoPublish",
  "$schema": "http://json-schema.org/schema",
  "title": "Publish for Expo",
  "description": "Deploy a project to Expo hosting",
  "type": "object",
  "properties": {
    "quiet": {
      "type": "boolean",
      "description": "Suppress verbose output from the Metro bundler",
      "default": false,
      "alias": "q"
    },
    "sendTo": {
      "type": "string",
      "description": "A phone number or email address to send a link to",
      "alias": "s"
    },
    "clear": {
      "type": "boolean",
      "description": "Clear the Metro bundler cache",
      "default": false,
      "alias": "c"
    },
    "target": {
      "enum": ["managed", "bare"],
      "default": "managed",
      "description": "Target environment for which this publish is intended. Options are managed or bare.",
      "alias": "t"
    },
    "maxWorkers": {
      "type": "number",
      "description": "Maximum number of tasks to allow Metro to spawn"
    },
    "releaseChannel": {
      "type": "string",
      "description": "The release channel to publish to. Default is 'default'.",
      "default": "default"
    }
  }
}
