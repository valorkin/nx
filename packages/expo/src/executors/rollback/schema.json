{
  "cli": "nx",
  "$id": "NxExpoRollback",
  "$schema": "http://json-schema.org/schema",
  "title": "Rollback Publish Command for Expo",
  "description": "Undo an update to a channel",
  "type": "object",
  "properties": {
    "releaseChannel": {
      "type": "string",
      "description": "The release channel to publish to."
    },
    "sdkVersion": {
      "type": "string",
      "description": "The sdk version to rollback."
    },
    "platform": {
      "enum": ["ios", "android"],
      "description": "The platform to rollback."
    }
  },
  "required": ["releaseChannel", "sdkVersion"]
}
