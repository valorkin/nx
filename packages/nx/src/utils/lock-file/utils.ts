import { defaultHashing } from '../../hasher/hashing-impl';

/**
 * Simple sort function to ensure keys are ordered alphabetically
 * @param obj
 * @returns
 */
export function sortObject<T = string>(
  obj: Record<string, T>,
  valueTransformator: (value: T) => any = (value) => value
): Record<string, T> | undefined {
  const keys = Object.keys(obj);
  if (keys.length === 0) {
    return;
  }

  const result: Record<string, T> = {};
  keys.sort().forEach((key) => {
    result[key] = valueTransformator(obj[key]);
  });
  return result;
}

/**
 * Apply simple hashing of the content using the default hashing implementation
 * @param fileContent
 * @returns
 */
export function hashString(fileContent: string): string {
  return defaultHashing.hashArray([fileContent]);
}
