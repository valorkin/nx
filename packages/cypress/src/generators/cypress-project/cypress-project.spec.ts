import {
  addProjectConfiguration,
  readJson,
  readProjectConfiguration,
  Tree,
  updateProjectConfiguration,
  WorkspaceJsonConfiguration,
} from '@nrwl/devkit';
import { createTreeWithEmptyV1Workspace } from '@nrwl/devkit/testing';
import { cypressProjectGenerator } from './cypress-project';
import { Schema } from './schema';
import { Linter } from '@nrwl/linter';
import { installedCypressVersion } from '../../utils/cypress-version';

jest.mock('../../utils/cypress-version');

describe('Cypress Project', () => {
  let tree: Tree;
  const defaultOptions: Omit<Schema, 'name' | 'project'> = {
    linter: Linter.EsLint,
    standaloneConfig: false,
  };
  let mockedInstalledCypressVersion: jest.Mock<
    ReturnType<typeof installedCypressVersion>
  > = installedCypressVersion as never;

  beforeEach(() => {
    tree = createTreeWithEmptyV1Workspace();

    addProjectConfiguration(tree, 'my-app', {
      root: 'my-app',
      targets: {
        serve: {
          executor: 'serve-executor',
          options: {},
          configurations: {
            production: {},
          },
        },
      },
    });

    addProjectConfiguration(tree, 'my-dir-my-app', {
      root: 'my-dir/my-app',
      targets: {
        serve: {
          executor: 'serve-executor',
          options: {},
          configurations: {
            production: {},
          },
        },
      },
    });
  });
  afterEach(() => jest.clearAllMocks());

  describe('> v10', () => {
    beforeEach(() => {
      mockedInstalledCypressVersion.mockReturnValue(10);
    });

    it('should generate files for v10 and above', async () => {
      await cypressProjectGenerator(tree, {
        ...defaultOptions,
        name: 'my-app-e2e',
        project: 'my-app',
      });

      expect(tree.exists('apps/my-app-e2e/cypress.config.ts')).toBeTruthy();

      expect(
        tree.exists('apps/my-app-e2e/src/fixtures/example.json')
      ).toBeTruthy();
      expect(tree.exists('apps/my-app-e2e/src/e2e/app.cy.ts')).toBeTruthy();
      expect(tree.exists('apps/my-app-e2e/src/support/app.po.ts')).toBeTruthy();
      expect(
        tree.exists('apps/my-app-e2e/src/support/commands.ts')
      ).toBeTruthy();
      expect(tree.exists('apps/my-app-e2e/src/support/e2e.ts')).toBeTruthy();
    });

    it('should add update `workspace.json` file properly when eslint is passed', async () => {
      await cypressProjectGenerator(tree, {
        name: 'my-app-e2e',
        project: 'my-app',
        linter: Linter.EsLint,
        standaloneConfig: false,
      });
      const workspaceJson = readJson(tree, 'workspace.json');
      const project = workspaceJson.projects['my-app-e2e'];

      expect(project.architect.lint).toMatchSnapshot();
    });

    it('should not add lint target when "none" is passed', async () => {
      await cypressProjectGenerator(tree, {
        name: 'my-app-e2e',
        project: 'my-app',
        linter: Linter.None,
        standaloneConfig: false,
      });
      const workspaceJson = readJson(tree, 'workspace.json');
      const project = workspaceJson.projects['my-app-e2e'];

      expect(project.architect.lint).toBeUndefined();
    });

    it('should update tags and implicit dependencies', async () => {
      await cypressProjectGenerator(tree, {
        name: 'my-app-e2e',
        project: 'my-app',
        linter: Linter.EsLint,
        standaloneConfig: false,
      });

      const project = readProjectConfiguration(tree, 'my-app-e2e');
      expect(project.tags).toEqual([]);
      expect(project.implicitDependencies).toEqual(['my-app']);
    });

    it('should set right path names in `cypress.config.ts`', async () => {
      await cypressProjectGenerator(tree, {
        ...defaultOptions,
        name: 'my-app-e2e',
        project: 'my-app',
      });
      const cypressConfig = tree.read(
        'apps/my-app-e2e/cypress.config.ts',
        'utf-8'
      );
      expect(cypressConfig).toMatchSnapshot();
    });

    it('should set right path names in `tsconfig.e2e.json`', async () => {
      await cypressProjectGenerator(tree, {
        ...defaultOptions,
        name: 'my-app-e2e',
        project: 'my-app',
      });
      const tsconfigJson = readJson(tree, 'apps/my-app-e2e/tsconfig.json');
      expect(tsconfigJson).toMatchSnapshot();
    });

    it('should extend from tsconfig.base.json', async () => {
      await cypressProjectGenerator(tree, {
        ...defaultOptions,
        name: 'my-app-e2e',
        project: 'my-app',
      });

      const tsConfig = readJson(tree, 'apps/my-app-e2e/tsconfig.json');
      expect(tsConfig.extends).toBe('../../tsconfig.base.json');
    });

    it('should support a root tsconfig.json instead of tsconfig.base.json', async () => {
      tree.rename('tsconfig.base.json', 'tsconfig.json');

      await cypressProjectGenerator(tree, {
        ...defaultOptions,
        name: 'my-app-e2e',
        project: 'my-app',
      });

      const tsConfig = readJson(tree, 'apps/my-app-e2e/tsconfig.json');
      expect(tsConfig.extends).toBe('../../tsconfig.json');
    });

    describe('nested', () => {
      it('should set right path names in `cypress.config.ts`', async () => {
        await cypressProjectGenerator(tree, {
          ...defaultOptions,
          name: 'my-app-e2e',
          project: 'my-dir-my-app',
          directory: 'my-dir',
        });

        const cypressConfig = tree.read(
          'apps/my-dir/my-app-e2e/cypress.config.ts',
          'utf-8'
        );
        expect(cypressConfig).toMatchSnapshot();
      });

      it('should set right path names in `tsconfig.e2e.json`', async () => {
        await cypressProjectGenerator(tree, {
          ...defaultOptions,
          name: 'my-app-e2e',
          project: 'my-dir-my-app',
          directory: 'my-dir',
        });
        const tsconfigJson = readJson(
          tree,
          'apps/my-dir/my-app-e2e/tsconfig.json'
        );

        expect(tsconfigJson).toMatchSnapshot();
      });

      it('should extend from tsconfig.base.json', async () => {
        await cypressProjectGenerator(tree, {
          ...defaultOptions,
          name: 'my-app-e2e',
          project: 'my-app',
          directory: 'my-dir',
        });

        const tsConfig = readJson(tree, 'apps/my-dir/my-app-e2e/tsconfig.json');
        expect(tsConfig.extends).toBe('../../../tsconfig.base.json');
      });

      it('should support a root tsconfig.json instead of tsconfig.base.json', async () => {
        tree.rename('tsconfig.base.json', 'tsconfig.json');

        await cypressProjectGenerator(tree, {
          ...defaultOptions,
          name: 'my-app-e2e',
          project: 'my-app',
          directory: 'my-dir',
        });

        const tsConfig = readJson(tree, 'apps/my-dir/my-app-e2e/tsconfig.json');
        expect(tsConfig.extends).toBe('../../../tsconfig.json');
      });
    });

    describe('--project', () => {
      describe('none', () => {
        it('should not add any implicit dependencies', async () => {
          await cypressProjectGenerator(tree, {
            ...defaultOptions,
            name: 'my-app-e2e',
            baseUrl: 'http://localhost:7788',
          });

          const workspaceJson = readJson<WorkspaceJsonConfiguration>(
            tree,
            'workspace.json'
          );
          const projectConfig = workspaceJson.projects['my-app-e2e'];
          expect(projectConfig.implicitDependencies).not.toBeDefined();
          expect(projectConfig.tags).toEqual([]);
        });
      });

      it('should not throw an error when --project does not have targets', async () => {
        const projectConf = readProjectConfiguration(tree, 'my-app');
        delete projectConf.targets;

        updateProjectConfiguration(tree, 'my-app', projectConf);
        await cypressProjectGenerator(tree, {
          name: 'my-app-e2e',
          project: 'my-app',
          linter: Linter.EsLint,
        });

        const projectConfig = readProjectConfiguration(tree, 'my-app-e2e');
        expect(projectConfig.targets['e2e'].options.devServerTarget).toEqual(
          'my-app:serve'
        );
      });
    });

    it('should generate in the correct folder', async () => {
      await cypressProjectGenerator(tree, {
        ...defaultOptions,
        name: 'other-e2e',
        project: 'my-app',
        directory: 'one/two',
      });
      const workspace = readJson(tree, 'workspace.json');
      expect(workspace.projects['one-two-other-e2e']).toBeDefined();
      [
        'apps/one/two/other-e2e/cypress.config.ts',
        'apps/one/two/other-e2e/src/e2e/app.cy.ts',
      ].forEach((path) => expect(tree.exists(path)).toBeTruthy());
    });
  });

  describe('v9 - v7', () => {
    beforeEach(() => {
      mockedInstalledCypressVersion.mockReturnValue(9);
    });
    it('should generate files', async () => {
      await cypressProjectGenerator(tree, {
        ...defaultOptions,
        name: 'my-app-e2e',
        project: 'my-app',
      });

      expect(tree.exists('apps/my-app-e2e/cypress.json')).toBeTruthy();

      expect(
        tree.exists('apps/my-app-e2e/src/fixtures/example.json')
      ).toBeTruthy();
      expect(
        tree.exists('apps/my-app-e2e/src/integration/app.spec.ts')
      ).toBeTruthy();
      expect(tree.exists('apps/my-app-e2e/src/support/app.po.ts')).toBeTruthy();
      expect(
        tree.exists('apps/my-app-e2e/src/support/commands.ts')
      ).toBeTruthy();
      expect(tree.exists('apps/my-app-e2e/src/support/index.ts')).toBeTruthy();
    });
  });

  describe('< v7', () => {
    beforeEach(() => {
      mockedInstalledCypressVersion.mockReturnValue(6);
    });

    it('should generate a plugin file if cypress is below version 7', async () => {
      await cypressProjectGenerator(tree, {
        ...defaultOptions,
        name: 'my-app-e2e',
        project: 'my-app',
      });

      expect(tree.exists('apps/my-app-e2e/src/plugins/index.js')).toBeTruthy();
    });

    it('should add update `workspace.json` file', async () => {
      await cypressProjectGenerator(tree, {
        name: 'my-app-e2e',
        project: 'my-app',
        linter: Linter.EsLint,
        standaloneConfig: false,
      });
      const workspaceJson = readJson(tree, 'workspace.json');
      const project = workspaceJson.projects['my-app-e2e'];

      expect(project.root).toEqual('apps/my-app-e2e');

      expect(project.architect).toMatchSnapshot();
    });

    it('should add update `workspace.json` file (baseUrl)', async () => {
      await cypressProjectGenerator(tree, {
        name: 'my-app-e2e',
        project: 'my-app',
        baseUrl: 'http://localhost:3000',
        linter: Linter.EsLint,
        standaloneConfig: false,
      });
      const workspaceJson = readJson(tree, 'workspace.json');
      const project = workspaceJson.projects['my-app-e2e'];

      expect(project.root).toEqual('apps/my-app-e2e');

      expect(project.architect).toMatchSnapshot();
    });

    it('should add update `workspace.json` file for a project with a defaultConfiguration', async () => {
      const originalProject = readProjectConfiguration(tree, 'my-app');
      originalProject.targets.serve.defaultConfiguration = 'development';
      originalProject.targets.serve.configurations.development = {};
      updateProjectConfiguration(tree, 'my-app', originalProject);

      await cypressProjectGenerator(tree, {
        name: 'my-app-e2e',
        project: 'my-app',
        linter: Linter.EsLint,
        standaloneConfig: false,
      });
      const workspaceJson = readJson(tree, 'workspace.json');
      const project = workspaceJson.projects['my-app-e2e'];

      expect(project.root).toEqual('apps/my-app-e2e');

      expect(project.architect).toMatchSnapshot();
    });

    describe('nested', () => {
      it('should update workspace.json', async () => {
        await cypressProjectGenerator(tree, {
          name: 'my-app-e2e',
          project: 'my-dir-my-app',
          directory: 'my-dir',
          linter: Linter.EsLint,
          standaloneConfig: false,
        });
        const projectConfig = readJson(tree, 'workspace.json').projects[
          'my-dir-my-app-e2e'
        ];

        expect(projectConfig).toBeDefined();
        expect(projectConfig.architect).toMatchSnapshot();
      });
    });

    describe('--linter', () => {
      describe('eslint', () => {
        it('should add eslint-plugin-cypress', async () => {
          await cypressProjectGenerator(tree, {
            name: 'my-app-e2e',
            project: 'my-app',
            linter: Linter.EsLint,
            standaloneConfig: false,
          });
          const packageJson = readJson(tree, 'package.json');
          expect(
            packageJson.devDependencies['eslint-plugin-cypress']
          ).toBeTruthy();

          const eslintrcJson = readJson(tree, 'apps/my-app-e2e/.eslintrc.json');
          expect(eslintrcJson).toMatchSnapshot();
        });
      });
    });

    describe('project with directory in its name', () => {
      beforeEach(async () => {
        await cypressProjectGenerator(tree, {
          name: 'my-dir/my-app-e2e',
          project: 'my-dir-my-app',
          linter: Linter.EsLint,
          standaloneConfig: false,
        });
      });

      it('should update workspace.json', async () => {
        const projectConfig = readJson(tree, 'workspace.json').projects[
          'my-dir-my-app-e2e'
        ];

        expect(projectConfig).toBeDefined();
        expect(projectConfig.architect).toMatchSnapshot();
      });

      it('should update nx.json', async () => {
        const project = readProjectConfiguration(tree, 'my-dir-my-app-e2e');
        expect(project.tags).toEqual([]);
        expect(project.implicitDependencies).toEqual(['my-dir-my-app']);
      });

      it('should set right path names in `cypress.json`', async () => {
        const cypressConfig = tree.read(
          'apps/my-dir/my-app-e2e/cypress.json',
          'utf-8'
        );

        expect(cypressConfig).toMatchSnapshot();
      });
    });
  });
});
