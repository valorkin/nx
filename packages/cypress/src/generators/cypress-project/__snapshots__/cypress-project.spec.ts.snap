// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cypress Project < v7 --linter eslint should add eslint-plugin-cypress 1`] = `
Object {
  "extends": Array [
    "plugin:cypress/recommended",
    "../../.eslintrc.json",
  ],
  "ignorePatterns": Array [
    "!**/*",
  ],
  "overrides": Array [
    Object {
      "files": Array [
        "*.ts",
        "*.tsx",
        "*.js",
        "*.jsx",
      ],
      "rules": Object {},
    },
    Object {
      "files": Array [
        "src/plugins/index.js",
      ],
      "rules": Object {
        "@typescript-eslint/no-var-requires": "off",
        "no-undef": "off",
      },
    },
  ],
}
`;

exports[`Cypress Project < v7 nested should update workspace.json 1`] = `
Object {
  "e2e": Object {
    "builder": "@nrwl/cypress:cypress",
    "configurations": Object {
      "production": Object {
        "devServerTarget": "my-dir-my-app:serve:production",
      },
    },
    "options": Object {
      "cypressConfig": "apps/my-dir/my-app-e2e/cypress.json",
      "devServerTarget": "my-dir-my-app:serve",
      "testingType": "e2e",
      "tsConfig": "apps/my-dir/my-app-e2e/tsconfig.json",
    },
  },
  "lint": Object {
    "builder": "@nrwl/linter:eslint",
    "options": Object {
      "lintFilePatterns": Array [
        "apps/my-dir/my-app-e2e/**/*.{js,ts}",
      ],
    },
    "outputs": Array [
      "{options.outputFile}",
    ],
  },
}
`;

exports[`Cypress Project < v7 project with directory in its name should set right path names in \`cypress.json\` 1`] = `
"{
  \\"fileServerFolder\\": \\".\\",
  \\"fixturesFolder\\": \\"./src/fixtures\\",
  \\"integrationFolder\\": \\"./src/integration\\",
  \\"modifyObstructiveCode\\": false,
  \\"supportFile\\": \\"./src/support/index.ts\\",
  \\"pluginsFile\\": \\"./src/plugins/index\\",
  \\"video\\": true,
  \\"videosFolder\\": \\"../../../dist/cypress/apps/my-dir/my-app-e2e/videos\\",
  \\"screenshotsFolder\\": \\"../../../dist/cypress/apps/my-dir/my-app-e2e/screenshots\\",
  \\"chromeWebSecurity\\": false
}
"
`;

exports[`Cypress Project < v7 project with directory in its name should update workspace.json 1`] = `
Object {
  "e2e": Object {
    "builder": "@nrwl/cypress:cypress",
    "configurations": Object {
      "production": Object {
        "devServerTarget": "my-dir-my-app:serve:production",
      },
    },
    "options": Object {
      "cypressConfig": "apps/my-dir/my-app-e2e/cypress.json",
      "devServerTarget": "my-dir-my-app:serve",
      "testingType": "e2e",
      "tsConfig": "apps/my-dir/my-app-e2e/tsconfig.json",
    },
  },
  "lint": Object {
    "builder": "@nrwl/linter:eslint",
    "options": Object {
      "lintFilePatterns": Array [
        "apps/my-dir/my-app-e2e/**/*.{js,ts}",
      ],
    },
    "outputs": Array [
      "{options.outputFile}",
    ],
  },
}
`;

exports[`Cypress Project < v7 should add update \`workspace.json\` file (baseUrl) 1`] = `
Object {
  "e2e": Object {
    "builder": "@nrwl/cypress:cypress",
    "options": Object {
      "baseUrl": "http://localhost:3000",
      "cypressConfig": "apps/my-app-e2e/cypress.json",
      "testingType": "e2e",
      "tsConfig": "apps/my-app-e2e/tsconfig.json",
    },
  },
  "lint": Object {
    "builder": "@nrwl/linter:eslint",
    "options": Object {
      "lintFilePatterns": Array [
        "apps/my-app-e2e/**/*.{js,ts}",
      ],
    },
    "outputs": Array [
      "{options.outputFile}",
    ],
  },
}
`;

exports[`Cypress Project < v7 should add update \`workspace.json\` file 1`] = `
Object {
  "e2e": Object {
    "builder": "@nrwl/cypress:cypress",
    "configurations": Object {
      "production": Object {
        "devServerTarget": "my-app:serve:production",
      },
    },
    "options": Object {
      "cypressConfig": "apps/my-app-e2e/cypress.json",
      "devServerTarget": "my-app:serve",
      "testingType": "e2e",
      "tsConfig": "apps/my-app-e2e/tsconfig.json",
    },
  },
  "lint": Object {
    "builder": "@nrwl/linter:eslint",
    "options": Object {
      "lintFilePatterns": Array [
        "apps/my-app-e2e/**/*.{js,ts}",
      ],
    },
    "outputs": Array [
      "{options.outputFile}",
    ],
  },
}
`;

exports[`Cypress Project < v7 should add update \`workspace.json\` file for a project with a defaultConfiguration 1`] = `
Object {
  "e2e": Object {
    "builder": "@nrwl/cypress:cypress",
    "configurations": Object {
      "production": Object {
        "devServerTarget": "my-app:serve:production",
      },
    },
    "options": Object {
      "cypressConfig": "apps/my-app-e2e/cypress.json",
      "devServerTarget": "my-app:serve:development",
      "testingType": "e2e",
      "tsConfig": "apps/my-app-e2e/tsconfig.json",
    },
  },
  "lint": Object {
    "builder": "@nrwl/linter:eslint",
    "options": Object {
      "lintFilePatterns": Array [
        "apps/my-app-e2e/**/*.{js,ts}",
      ],
    },
    "outputs": Array [
      "{options.outputFile}",
    ],
  },
}
`;

exports[`Cypress Project > v10 nested should set right path names in \`cypress.config.ts\` 1`] = `
"import { defineConfig } from 'cypress';
import { nxE2EPreset } from '@nrwl/cypress/plugins/cypress-preset';

export default defineConfig({
  e2e: nxE2EPreset(__dirname)
});
"
`;

exports[`Cypress Project > v10 nested should set right path names in \`tsconfig.e2e.json\` 1`] = `
Object {
  "compilerOptions": Object {
    "allowJs": true,
    "outDir": "../../../dist/out-tsc",
    "sourceMap": false,
    "types": Array [
      "cypress",
      "node",
    ],
  },
  "extends": "../../../tsconfig.base.json",
  "include": Array [
    "src/**/*.ts",
    "src/**/*.js",
    "cypress.config.ts",
  ],
}
`;

exports[`Cypress Project > v10 should add update \`workspace.json\` file properly when eslint is passed 1`] = `
Object {
  "builder": "@nrwl/linter:eslint",
  "options": Object {
    "lintFilePatterns": Array [
      "apps/my-app-e2e/**/*.{js,ts}",
    ],
  },
  "outputs": Array [
    "{options.outputFile}",
  ],
}
`;

exports[`Cypress Project > v10 should set right path names in \`cypress.config.ts\` 1`] = `
"import { defineConfig } from 'cypress';
import { nxE2EPreset } from '@nrwl/cypress/plugins/cypress-preset';

export default defineConfig({
  e2e: nxE2EPreset(__dirname)
});
"
`;

exports[`Cypress Project > v10 should set right path names in \`tsconfig.e2e.json\` 1`] = `
Object {
  "compilerOptions": Object {
    "allowJs": true,
    "outDir": "../../dist/out-tsc",
    "sourceMap": false,
    "types": Array [
      "cypress",
      "node",
    ],
  },
  "extends": "../../tsconfig.base.json",
  "include": Array [
    "src/**/*.ts",
    "src/**/*.js",
    "cypress.config.ts",
  ],
}
`;
